### ISSUE TITLE
General improvements to woocommerce/components

### ISSUE THREAD

**Issue by samueljseay** [woocommerce/woocommerce#32246]

**Comment by mattsherman** [comment]
@joelclimbsthings What is the status of this? You were exploring the css-in-js solution, right? A P2 post with your findings would be excellent. And we should put the rest of his on the backburner until the next cool down (we can also consider some of this work for a future cycle).

**Comment by joelclimbsthings** [comment]
@mattsherman Yup! I spent some time looking into alternatives to Linaria as well. Planning to put up on a P2, although I didn't get as far with this as I'd like due to portering and other tasks, so I plan to finish it off during the next cool down.

**Comment by timmy5685** [comment]
Hi @samueljseay - thanks for putting this together. We're going through some older issues to sort out what's still relevant, and since this one's a 2021 epic covering four separate threads, it's worth a status pass on each:

- TypeScript conversion + published type defs — largely done. `@woocommerce/components` is ~63% TS now (199 TS vs 119 JS source files), and it publishes type defs (`"types": "build-types/index.d.ts"`, built via a dedicated `tsc --emitDeclarationOnly` step). Goal met and then some.

- css-in-js / Linaria spike — didn't happen. There are no `linaria` references left anywhere in the repo and the package is still SCSS-based; the spike stalled in 2021 and never came back.

- Longform MDX docs — didn't happen. Storybook is still there (66 stories for the package), but no MDX docs alongside them.

- Size reduction — no dedicated effort landed, though the package does now ship ESM and declares `sideEffects` for tree-shaking, so consumer-side reduction is at least enabled.

Given that, I don't think this epic is tracking live work anymore — one thread finished organically, one was dropped, and two never shipped. Going to close it as stale. The only genuinely open piece is finishing the TS conversion (~119 legacy JS files, mostly the d3 charts / `advanced-filters` / calendar), and that's better off as its own scoped issue if there's appetite — happy to open one.
